assets/app/scripts/controllers/build.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PodController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('BuildController', function ($scope, $routeParams, DataService, project, BuildsService, $filter) {
+    $scope.build = null;
+    $scope.buildConfigName = $routeParams.buildconfig;
+    $scope.builds = {};
+    $scope.alerts = {};
+    $scope.renderOptions = $scope.renderOptions || {};    
+    $scope.renderOptions.hideFilterWidget = true;    
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig,
+        link: "project/" + $routeParams.project + "/browse/builds/" + $routeParams.buildconfig
+      },      
+      {
+        title: $routeParams.build
+      }
+    ];
+
+    var watches = [];
+    var objectWatches = [];
+
+    project.get($routeParams.project).then(function(resp) {
+      angular.extend($scope, {
+        project: resp[0],
+        projectPromise: resp[1].projectPromise
+      });
+      DataService.get("builds", $routeParams.build, $scope).then(
+        // success
+        function(build) {
+          $scope.build = build;
+          var buildNumber = $filter("annotation")(build, "buildNumber");
+          if (buildNumber) {
+            $scope.breadcrumbs[2].title = "#" + buildNumber;
+          }
+
+          // If we found the item successfully, watch for changes on it
+          objectWatches.push(DataService.watchObject("builds", $routeParams.build, $scope, function(build, action) {
```

Comment:
Is it terrible to just keep them separate? Have one web socket that watches the list and another that watches the single item? How common is this? It seems we're optimizing for the edge case.

If you have a thousand pods in your namespace and only care about one, you're going to get a lot of messages you just throw away.